core/lib/spree/events/processors/mail_processor.rb

```diff
@@ -0,0 +1,54 @@
+module Spree
+  module Events
+    module Processors
+      class MailProcessor
+        class << self
+          # This subscribes the MailProcessor to the relevant events in Solidus.
+          # It is called in an initializer BEFORE events start happening.
+          #
+          # To override/modify which notifications are sent, add a decorator for
+          # this file to define your own `#register!` method. For example:
+          #
+          # Spree::Events::Processors::MailProcessor.class_eval do
```

Comment:
If we plan to move this class into an action mailer extension anyway and people are able to write their own mail processors we do not need to imply `class_eval` is a good idea to use the event bus. This violates the whole purpose of an event bus, no?